dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/IndexServiceBean.java

```diff
@@ -960,21 +960,24 @@ private String addOrUpdateDataset(IndexableDataset indexableDataset) {
                     if (fileMetadata != null) {
                         String filenameComplete = fileMetadata.getLabel();
                         if (filenameComplete != null) {
-                            String filenameWithoutExtension = "";
-                            // String extension = "";
                             int i = filenameComplete.lastIndexOf('.');
                             if (i > 0) {
-                                // extension = filenameComplete.substring(i + 1);
                                 try {
-                                    filenameWithoutExtension = filenameComplete.substring(0, i);
+                                    String fileExtension = filenameComplete.substring(i + 1);
```

Comment:
I wonder if there are any Apache Commons IO or even JDK classes (i.e. `Path`?) that separate file name from extension reliably?

Reply:
There probably are, but I didn't want to change 'logic' behind how they substract file name. 
Valid file names in Dataverse are way less strict that what a library would allow. 
Maybe we can think about it, but I think it would require us to talk with business about it.